### PR TITLE
Ra 579

### PR DESCRIPTION
**Pull request by ylatuya**

### CODE REVIEW

File: VAS.Core/MVVMC/ViewModelBase.cs [hunk 1]
```diff
@@ -36,6 +36,37 @@ public class ViewModelBase<T> : BindableBase, IViewModel<T> where T : INotifyPro
 			get;
 		}
 
+		/// <summary>
+		/// Checks if sync is required from the name of the porperty that triggered a <see cref="PropertyChangedEventHandler"/>.
```
Comment (vmartos) [inline VAS.Core/MVVMC/ViewModelBase.cs]: typo "porperty"

File: VAS.Core/MVVMC/ViewModelBase.cs [hunk 2]
```diff
@@ -36,6 +36,37 @@ public class ViewModelBase<T> : BindableBase, IViewModel<T> where T : INotifyPro
...
+		/// <returns><c>true</c>, if sync was needed, <c>false</c> otherwise.</returns>
+		/// <param name="propertyNameChanged">Property name changed.</param>
+		/// <param name="propertyNameToCheck">Property name to check.</param>
+		public bool NeedsSync (string propertyNameChanged, string propertyNameToCheck)
+		{
+			return NeedsSync (propertyNameChanged, propertyNameToCheck, null, null);
+		}
+
+		/// <summary>
+		/// Checks if sync is required from the name of the porperty that triggered a <see cref="PropertyChangedEventHandler"/>
```
Comment (vmartos) [inline VAS.Core/MVVMC/ViewModelBase.cs]: typo "porperty"

File: VAS.Core/MVVMC/ViewModelBase.cs [hunk 3]
```diff
@@ -36,6 +36,37 @@ public class ViewModelBase<T> : BindableBase, IViewModel<T> where T : INotifyPro
...
+		/// and the sender of the event.
+		/// </summary>
+		/// <returns><c>true</c>, if sync was needed, <c>false</c> otherwise.</returns>
+		/// <param name="sender">Sender.</param>
+		/// <param name="senderToCheck">Sender to check.</param>
+		public bool NeedsSync (string propertyNameChanged, string propertyNameToCheck, object sender, object senderToCheck)
+		{
+			if (propertyNameChanged == null) {
+				return true;
```
Comment (vmartos) [inline VAS.Core/MVVMC/ViewModelBase.cs]: why return true if propertyNameChanged is null?

File: VAS.Core/MVVMC/StorableVM.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+				if (Model.IsLoaded) {
+					SyncLoadedModel ();
+				}
+			}
+		}
+
+		/// <summary>
+		/// Synchronizes the loaded Model properties with the ViewModel.
+		/// </summary>
+		protected virtual void SyncLoadedModel ()
```
Comment (vmartos) [inline VAS.Core/MVVMC/StorableVM.cs]: as it haven't any implementation, could it be abstract?

File: VAS.Core/MVVMC/ViewModelBase.cs (hunk 3 above)
Comment (ylatuya) [inline VAS.Core/MVVMC/ViewModelBase.cs]: It's a way to resyncrhonize all properties, for examples when you setup a Model for the first time.

File: VAS.Core/MVVMC/StorableVM.cs (hunk 4 above)
Comment (ylatuya) [inline VAS.Core/MVVMC/StorableVM.cs]: The implementation is not required, for example for a ViewModel that have all Preloaded properties or a ViewModel that don't have any Perloaded property.

**Pull request merged**